starter-code/HeterogeneousRecyclerView/app/src/main/java/com/charlesdrews/heterogeneousrecyclerview/FootballRecyclerViewAdapter.java

```diff
@@ -17,6 +18,48 @@
 
 public class FootballRecyclerViewAdapter extends RecyclerView.Adapter {
 
+    private List<BaseFootballObject> mBaseFootballObjects;
+
+    public FootballRecyclerViewAdapter(List<BaseFootballObject> listOfFutbol) {
+        mBaseFootballObjects = listOfFutbol;
+    }
+
+    @Override
+    public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        if (viewType == 0){
+            View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.team_list_entry, null);
+            return new TeamViewHolder(view);
+        } else{
+            View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.player_list_entry, null);
+            return new PlayerViewHolder(view);
+        }
+
+    }
+
+    @Override
+    public void onBindViewHolder(RecyclerView.ViewHolder holder, int position) {
+        BaseFootballObject baseFootballObject = mBaseFootballObjects.get(holder.getAdapterPosition());
+        if (Team.class.equals(baseFootballObject.getClass())){
```

Comment:
This approach of comparing classes works, but is slow. A better approach would be to call the `holder.getItemViewType()` method to get the int assigned to the holder by your overridden `getItemViewType()` below, then check if it's 0 or 1.